// Copyright 2017 The CrunchyCrypt Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package crunchy;
option java_package = "com.google.security.crunchy.internal.keyset.testdata";

import "crunchy/key_management/internal/keyset.proto";

option java_outer_classname = "FactoryTestVectorsOuterClass";
option java_multiple_files = true;

message CrypterFactoryTestVector {
  Keyset keyset = 1;
  bytes aad = 2;
  bytes plaintext = 3;
  bytes ciphertext = 4;
}

message CrypterFactoryTestVectors {
  repeated CrypterFactoryTestVector test_vector = 1;
}

message MacerFactoryTestVector {
  Keyset keyset = 1;
  bytes message = 2;
  bytes signature = 3;
}

message MacerFactoryTestVectors {
  repeated MacerFactoryTestVector test_vector = 1;
}

message SignerFactoryTestVector {
  Keyset public_keyset = 1;
  Keyset private_keyset = 2;
  bytes message = 3;
  bytes signature = 4;
}

message SignerFactoryTestVectors {
  repeated SignerFactoryTestVector test_vector = 1;
}

message HybridCrypterFactoryTestVector {
  Keyset public_keyset = 1;
  Keyset private_keyset = 2;
  bytes plaintext = 3;
  bytes ciphertext = 4;
}

message HybridCrypterFactoryTestVectors {
  repeated HybridCrypterFactoryTestVector test_vector = 1;
}
